### DISCUSSION TITLE
Umzug's CLI API lacks consideration for DB connection lifecycle

### DISCUSSION THREAD

**Discussion by Hornwitser** [sequelize/umzug#622] [General]
In Sequelize you create a database connection with

```ts
const sequelize = new Sequelize(uri);
```

And tear it down with

```ts
await sequelize.close();
```

And until the connection has been torn down it may hold timers and connections in the event loop, and thefore keep the Node.js application running for however long those timers and connections take to time out.

Despite this being how most software talking to a database work Umzug's CLI API seems to just completely ignore this? The examples using Sequelize don't call `.close()`. The most sensible option I found to implement closing the database after being done with it is to export a `close` function in the definition of the umzug migrations

```ts
export async function close() {
    await sequelize.close();
}
```

and then call this after runAsCLI

```ts
import { migrator, close } from "./umzug.js"
await migrator.runAsCLI();
await close();
```

Another pain point is that to create an `Umzug` migrator with a `SequelizeStorage` adapter it needs to be passed a `Sequelize` instance which in turn needs a valid database URI. But if you are a user of this tool and want to see the `--help` text, getting an error telling you you're missing the database URI might make you think there is no `--help` and that passing the database credentials will make it execute the migrations.

The CLI should not need a database connection to show its `--help` text, and it should have a sensible place for tearing down the database connection after it's done with it. Exactly how these things are to be implemented I don't have a good idea for. The `beforeCommand` and `afterCommand` are a candidate as they don't get called when passing `--help`, but currently there's no way to set the storage and context from them.

**Comment by mmkal** [comment]
It's worth noting a couple of things:

- technically umzug is not a CLI, it just has a method to turn _your_ application into a CLI
- sequelize and its lifecycle is not managed by umzug at all. `SequelizeStorage` is really just a (supported) example storage. umzug doesn't know what you did with your `Sequelize` instance before creating an `Umzug` instance, and it doesn't know what you'll do with it afterwards. Maybe you're calling `await umzug.up(...)` as part of server route handler

That said, I agree that in practice your use case is common, and umzug considers sequelize a first-class citizen, so it would be nice to have something either built in, or easy to configure and well-documented for this scenario.
